twisted/twisted/plugins/tribler_plugin.py

```diff
@@ -0,0 +1,94 @@
+"""
+This twistd plugin enables to start Tribler headless using the twistd command.
+"""
+import logging
+import os
+import signal
+from time import sleep
+
+from twisted.application.service import MultiService, IServiceMaker
+from twisted.conch import manhole_tap
+from twisted.internet import reactor
+from twisted.plugin import IPlugin
+from twisted.python import usage
+from twisted.python.log import msg
+from zope.interface import implements
+
+from Tribler.Core.Session import Session
+from Tribler.Core.SessionConfig import SessionStartupConfig
+
+
+class Options(usage.Options):
+    optParameters = [
+        ["manhole", "m", 0, "Enable manhole telnet service listening at the specified port", int],
+    ]
+
+
+class TriblerServiceMaker(object):
+    implements(IServiceMaker, IPlugin)
+    tapname = "tribler"
+    description = "Tribler twistd plugin, starts Tribler as a twistd process"
+    options = Options
+
+    def __init__(self):
+        """
+        Initialize the variables of the TriblerServiceMaker and the logger.
+        """
+        self.session = None
+        self._stopping = False
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+    def start_tribler(self):
+        """
+        Main method to startup Tribler.
+        """
+
+        def signal_handler(sig, _):
+            msg("Received shut down signal %s" % sig)
+            if not self._stopping:
+                self._stopping = True
+                try:
+                    self.session.shutdown()
+                    msg("Tribler shut down")
+                except Exception, exc:
```

Comment:
I've removed the try/catch block. We are not catching exceptions during Tribler shutdown in other files (`tribler_main.py`, `test_as_server.py`, `tunnel/main.py` etc).